ui/src/main/webapp/wizards/sources/graphql-queries/source-discovery.js

```diff
@@ -0,0 +1,221 @@
+import { gql } from 'react-apollo'
+import { getFriendlyMessage } from 'graphql-errors'
+
+const payloadFragments = {
+  cswConfigurationPayload: gql`
+    fragment cswSourceConfigurationPayload on CswSourceConfigurationPayload {
+      cswProfile
+      sourceName
```

Comment:
Not sure if it is worth it, but the `sourceName`, `endpointUrl`, `pid`, and `creds` in a separate fragment since these should be common among all sources.